pkg/eval/pwd_test.go

```diff
@@ -0,0 +1,40 @@
+package eval
+
+import (
+	"errors"
+	"path/filepath"
+	"testing"
+)
+
+func TestBuiltinPwd(t *testing.T) {
+	tmpHome, cleanup := InTempHome()
+	defer cleanup()
+
+	mustMkdirAll("dir1")
+	mustMkdirAll("dir2")
+	dir1 := filepath.Join(tmpHome, "dir1")
+	dir2 := filepath.Join(tmpHome, "dir2")
+
+	Test(t,
+		That(`pwd=dir1 put $pwd; put $pwd`).Puts(dir1, tmpHome),
+		That(`cd $E:HOME/dir2; pwd=~ put $pwd; put $pwd`).Puts(tmpHome, dir2),
+		That(`cd $E:HOME/dir2; pwd=~/dir1 put $pwd; put $pwd`).Puts(dir1, dir2),
+		That(`cd $E:HOME/dir1; pwd=../dir2 put $pwd; put $pwd`).Puts(dir2, dir1),
+		That(`pwd=(float64 1) put $pwd`).Throws(ErrPathMustBeString, "pwd=(float64 1)"),
+	)
+}
+
+// Verify the behavior when the CWD cannot be determined.
+func TestBuiltinPwdNoCwd(t *testing.T) {
```

Comment:
`TestBuiltinPwd_GetwdError` would be a better name.

(Names of test functions are one of the few places in idiomatic Go code that allow underscores.)

Reply:
done